### PR TITLE
[WIP] Prometheus Object Builder

### PR DESCRIPTION
**Pull request by umangachapagain**
Includes a script and mixin to convert Prometheus Alert files to PrometheusRule Object.
Ideally, this has to be integrated with Gluster Operator.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+echo 'Generating prometheus_alerts.yaml'
+jsonnet  -S -e 'std.manifestYamlDoc((import "alert-object.libsonnet").prometheus.rules)' > prometheus_alerts.yaml
```
Comment (cloudbehl) [inline build.sh]: can you document this in README file

File: rule-object.libsonnet [hunk 2]
```diff
@@ -0,0 +1,31 @@
+local mi = import 'mixin.libsonnet';
```
Comment (cloudbehl) [inline rule-object.libsonnet]: can you create this file under extras folder?

File: alert-object.libsonnet [hunk 3]
```diff
@@ -0,0 +1,31 @@
+local mi = import 'mixin.libsonnet';
```
Comment (cloudbehl) [inline alert-object.libsonnet]: can you move this file to extras folder?

**Comment by cloudbehl** [comment]
@umangachapagain The plan here is to merge this file object files into extras so we can create files which can be applied directly to k8s/oc.

File: build.sh (hunk 1 above)
Comment (umangachapagain) [reply build.sh]: this is not the recommended prcedure. 
Are you sure about documenting it in `master`? Maybe we can add a doc in `extras`?

File: rule-object.libsonnet (hunk 2 above)
Comment (umangachapagain) [reply rule-object.libsonnet]: will do.

File: alert-object.libsonnet (hunk 3 above)
Comment (umangachapagain) [reply alert-object.libsonnet]: will do.

**Comment by umangachapagain** [comment]
> @umangachapagain The plan here is to merge this file object files into extras so we can create files which can be applied directly to k8s/oc.

Understood. Will make the necessary changes.

**Comment by cloudbehl** [comment]
Closing this: Taken care as part of this PR https://github.com/gluster/gluster-mixins/pull/28

**Pull request closed without merging**